scaleway/resource_vpc_public_gateway_dhcp_reservation.go

```diff
@@ -0,0 +1,206 @@
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The configuration creation date.",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The configuration last modification date.",
+			},
+			"zone": zoneSchema(),
+		},
+	}
+}
+
+func resourceScalewayVPCPublicGatewayDHCPCReservationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcgwAPI, zone, err := vpcgwAPIWithZone(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	ip := net.ParseIP(d.Get("ip_address").(string))
+	if ip == nil {
+		return diag.FromErr(fmt.Errorf("could not parse ip_address"))
+	}
+
+	macAddress, err := net.ParseMAC(d.Get("mac_address").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	gatewayNetworkID := expandID(d.Get("gateway_network_id"))
+	_, err = vpcgwAPI.WaitForGatewayNetwork(&vpcgw.WaitForGatewayNetworkRequest{
+		GatewayNetworkID: gatewayNetworkID,
+		Zone:             zone,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	res, err := vpcgwAPI.CreateDHCPEntry(&vpcgw.CreateDHCPEntryRequest{
+		Zone:             zone,
+		MacAddress:       macAddress.String(),
+		IPAddress:        ip,
+		GatewayNetworkID: gatewayNetworkID,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(newZonedIDString(zone, res.ID))
+
+	_, err = vpcgwAPI.WaitForGatewayNetwork(&vpcgw.WaitForGatewayNetworkRequest{
+		Zone:             zone,
+		GatewayNetworkID: gatewayNetworkID,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return resourceScalewayVPCPublicGatewayDHCPReservationRead(ctx, d, meta)
+}
+
+func resourceScalewayVPCPublicGatewayDHCPReservationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcgwAPI, zone, ID, err := vpcgwAPIWithZoneAndID(meta, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	entry, err := vpcgwAPI.GetDHCPEntry(&vpcgw.GetDHCPEntryRequest{
+		DHCPEntryID: ID,
+		Zone:        zone,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		if is404Error(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	_ = d.Set("ip_address", entry.IPAddress.String())
+	_ = d.Set("mac_address", entry.MacAddress)
+	_ = d.Set("hostname", entry.Hostname)
+	_ = d.Set("type", entry.Type.String())
+	_ = d.Set("gateway_network_id", newZonedIDString(zone, entry.GatewayNetworkID))
+	_ = d.Set("created_at", entry.CreatedAt.Format(time.RFC3339))
+	_ = d.Set("updated_at", entry.UpdatedAt.Format(time.RFC3339))
+	_ = d.Set("zone", zone)
+
+	return nil
+}
+
+func resourceScalewayVPCPublicGatewayDHCPReservationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcgwAPI, zone, ID, err := vpcgwAPIWithZoneAndID(meta, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if d.HasChangesExcept("ip_address") {
```

Comment:
Why is the mac address never specified here? Why does ip_address as a special treatment?

Reply:
The API is not allowing this field to [update](https://github.com/scaleway/scaleway-sdk-go/blob/master/api/vpcgw/v1/vpcgw_sdk.go#L1790).